Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Search/OneWay.hs

```diff
@@ -192,21 +203,25 @@ oneWaySearch personId req bundleVersion clientVersion device = do
             gatewayUrl = merchant.gatewayUrl,
             searchRequestExpiry = searchRequest.validTill,
             customerLanguage = searchRequest.language,
+            city = originCity,
             disabilityTag = tag,
             device,
             shortestRouteInfo,
             ..
           }
   fork "updating search counters" $ do
-    merchantConfigs <- QMC.findAllByMerchantId person.merchantId
+    merchantConfigs <- QMC.findAllByMerchantOperatingCityId person.merchantOperatingCityId
     SMC.updateSearchFraudCounters personId merchantConfigs
-    mFraudDetected <- SMC.anyFraudDetected personId person.merchantId merchantConfigs
+    mFraudDetected <- SMC.anyFraudDetected personId merchantOperatingCity.id merchantConfigs
     whenJust mFraudDetected $ \mc -> SMC.blockCustomer personId (Just mc.id)
   return dSearchRes
   where
-    validateServiceability geoConfig =
-      unlessM (rideServiceable geoConfig someGeometriesContain req.origin.gps (Just req.destination.gps)) $
-        throwError RideNotServiceable
+    validateServiceability origin destination person' = do
+      originServiceability <- Serviceability.checkServiceability (.origin) (person'.id, person'.merchantId) origin False
+      destinationServiceability <- Serviceability.checkServiceability (.destination) (person'.id, person'.merchantId) destination False
+      if originServiceability.serviceable && destinationServiceability.serviceable
```

Comment:
Lets also compare in both cases city should be same else throw error InterCity ride is allowed.

Reply:
Added check for this.